### PR TITLE
assert issues

### PR DESCRIPTION
**Pull request by graebm**
- Add missing `#include <assert.h>` to enum-handling cpp files.
- Change assert() -> AWS_ASSERT() in input-validating discovery code.
- Run formatter on discovery code.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jobs/source/RejectedErrorCode.cpp [hunk 1]
```diff
@@ -18,6 +18,8 @@
 #include <aws/crt/StlAllocator.h>
 #include <aws/crt/StringUtils.h>
 
+#include <assert.h>
```
Comment (justinboswell) [inline jobs/source/RejectedErrorCode.cpp]: Why is this required? Why not just use AWS_ASSERT?

Comment (graebm) [reply jobs/source/RejectedErrorCode.cpp]: When transforming a string -> enum, if we don't recognize the string that the service returned, it does `assert(0); return static_cast<JobStatus>(-1);`

It's not something we want to be fatal in a release product. I asked Jonathan if he wanted the assert removed, or preserved and he wanted it preserved here.

Sorry for merging before I noticed this comment

Comment (Kevin-Tacheny) [reply jobs/source/RejectedErrorCode.cpp]: /home/kevin/deviceagentcpp-build/aws-iot-device-sdk-cpp-v2-prefix/src/aws-iot-device-sdk-cpp-v2/discovery/source/DiscoveryClient.cpp:32:47: error: ‘AWS_ASSERT’ was not declared in this scope

Still an issue I believe

Comment (graebm) [reply jobs/source/RejectedErrorCode.cpp]: This ought to work, the include chain is: `DiscoveryClient.cpp` -> `#include <aws/crt/Types.h>` -> `#include <aws/common/common.h>` -> `#include <aws/common/assert.inl>` ->`#define AWS_ASSERT`

Have you updated to [aws-crt-cpp v0.4.0](https://github.com/awslabs/aws-crt-cpp/releases/tag/v0.4.0)?

**Comment by Kevin-Tacheny** [comment]
Yeah I think you are correct I had some mismatch and now on to the next issue.  I am still stuck on this linker error in our code: https://github.com/awslabs/aws-crt-cpp/issues/26
